### PR TITLE
Feature/backEnd

### PR DESCRIPTION
**Pull request by arianacofone**
Hi guys!

This is a beast of a pull request, but it essentially sets up the following:
1. Umbrella Communication & File Structure for the Back-End
2. Register/Login/Authorization Functionality (From React to Database)
3. Login/Register Functionality in Browser for Users

There's two Trello cards this tackled:
[TrelloOne](https://trello.com/c/MBnFXg31/37-as-a-user-i-want-my-username-information-stored-into-a-database)
[TrelloTwo](https://trello.com/c/f8xE1eZY/24-as-new-i-want-to-login-register)

And here's some screen shots of this puppy in action:
Register/Login
image

Dashboard upon Registering/Login
image

User Table on Database (the last user is the example one from above)
image

I'm guessing you are going to have a good amount of questions, so send them over and thanks for your help!

### CODE REVIEW

File: client/components/member/Login.jsx [hunk 1]
```diff
@@ -1,13 +1,59 @@
...
+          />
+          </form>
       </div>
     );
   }
 }
 
 export default Login;
+
+Login.propTypes = propTypes;
```
Comment (nferner) [inline client/components/member/Login.jsx:57]: style - put "Login.propTypes = propTypes; before exporting

Comment (arianacofone) [reply client/components/member/Login.jsx:57]: Good catch! Just updated the login and register components to have the propTypes before the export.

File: client/components/member/Register.jsx [hunk 2]
```diff
@@ -1,13 +1,67 @@
...
+          />
+        </form>
       </div>
     );
   }
 }
 
 export default Register;
+
+Register.propTypes = propTypes;
```
Comment (nferner) [inline client/components/member/Register.jsx:65]: move register.propTypes above the export

Comment (arianacofone) [reply client/components/member/Register.jsx:65]: Super! It should be corrected now

File: client/components/member/Login.jsx (hunk 1 above)
Comment (nferner) [reply client/components/member/Login.jsx:57]: thanksh! didn't see anything else, made sense to me as you were making changes. You've got a plus one from me.

File: server.js [hunk 3]
```diff
@@ -2,10 +2,33 @@ require('dotenv').config();
...
+
+app.use(express.static(path.join(__dirname, '/dist')));
+app.get('/', (request, response) => {
+  response.sendFile(path.join(__dirname, 'dist/index.html'));
+});
+
+const port = process.env.PORT;
+app.listen(port, () => {
+  console.log(`Listening on Port ${port}`);
+});
```
Comment (tbongiorno1) [inline server.js:34]: Genuinely did not see anything off. Great Job A-money.

Comment (arianacofone) [reply server.js:34]: Phew! Thanks for reviewing :)

**Pull request merged**